### ISSUE TITLE
Debian UsrMerge

### ISSUE THREAD

**Issue by bt90** [syncthing/syncthing#9621]
### What happened?

Our current deb installs systemd service definitions in `/lib` which is basically deprecated:

```
/lib/systemd/system/syncthing@.service
/lib/systemd/system/syncthing-resume.service
```

We should move them to `/usr/lib` instead.

Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1073651

https://wiki.debian.org/UsrMerge

### Syncthing version

v1.27.9

### Platform & operating system

Linux

**Comment by imsodin** [comment]
Actually I am not entirely sure this is the right time for us to do this, or if we should do it at all.

tl;dr: I don't see relevant benefits and have a (admittedly diffuse) suspicion a change might do non-ideal things, thus I'd tend to leave the location as is.

As to the right time: The debian usr-merge transition is somewhat involved, to put it nicely - I wouldn't advice to go down the rabbit hole of why. In any case part of it was a full moratorium on moving any files between / and /usr. Now in the testing phase before the next stable release, they are doing moves in some very controlled way - hence e.g. the bug linked. As that bug says, it's a debian packaging thing though. There are e.g. mechanisms in place such that files don't get moved in current stable/backports. That's why I am unsure if this is the right time to do it on our end. Asking debian might be an option.

As to doing at all: The end state is /lib symlinked to /usr/lib. So if we have it in one or the other makes no difference on the filesystem. So in the absence of some concrete reason to move, maybe lets just not to avoid any complications of doing so (just random thought: fpm realizes the old file is gone, puts the new file in place and then delete the old file - ups no file there anymore, as both are the same).

And just for the record, this isn't a debian specific thing, quite the contrary, as usual they are some of the last ones to do it (still in progress and it was almost aborted at some point :) ). For the record some other docs on it:
https://fedoraproject.org/wiki/Features/UsrMove
https://www.freedesktop.org/wiki/Software/systemd/TheCaseForTheUsrMerge/

**Comment by bt90** [comment]
Isn't the symlink already in place these days? At least on my Ubuntu 24.04 installation:

```
/bin -> /usr/bin
/lib -> /usr/lib
```

**Comment by imsodin** [comment]
Yes it is, also on some/most debian ones, but not all. That's basically where the debian rabbithole starts.

Still my current knowledge leads to the opinion of not changing anything, as if the symlink is in place, it doesn't matter where it is, and if it's debian it's complicated so let them deal with it.

@bt90 Do you see that differently?

**Comment by bt90** [comment]
:+1:
